### PR TITLE
feat: merge package env vars

### LINKED ISSUE #951: Pepr helm chart does not add `env` from `package.json`
### Environment

N/A, local build only.

Definition of done:
- [ ] all Pepr settings from package.json are copied to values.yaml
- [ ] check to see if `PEPR_PRETTY_LOG` is changed if it breaks monitor command

### Steps to reproduce
1. Create a pepr capability with some `env` values in `package.json`.
2. `npx pepr build`
3. Compare the manifest against the helm chart values.

### Expected result

`env`s specified in `package.json` show up in both helm chart and manifest.

### Actual Result

Specified `env`s only appear in the manifest. Chart values have the default pepr envs.

### Visual Proof (screenshots, videos, text, etc)

Left side is manifest, right side is helm values:

Screenshot 2024-07-11 at 1 58 53 PM

### Severity/Priority

Low/medium - there are options to workaround this by adding more helm values but the mismatch in behavior is unexpected.

### Additional Context

Part of the reason I am afraid of using `env` helm values additions is a desire to not need to keep default envs in sync across updates since lists in helm will entirely override rather than appending. If Pepr structured the chart slightly differently to support `additionalEnv` or a map rather than a list that would also be a useful addition (can make this a separate issue for enhancements if needed).

### PR DESCRIPTION
**Pull request by cmwylie19**
## Description

After hearing from our users there is a need to merge `package.json` Pepr env vars into the `values.yaml` 

## Related Issue

Fixes #951

Relates to #

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://docs.pepr.dev/main/contribute/#submitting-a-pull-request) followed

### CODE REVIEW

File: journey/pepr-build.ts [hunk 1]
```diff
@@ -23,6 +24,72 @@ export function peprBuild() {
...
+        "name": "ZARF_VAR",
+        "value": "###ZARF_VAR_THING###"
+      }
+    ]
+
+    try {
+      const valuesYaml = await fs.readFile(resolve(cwd, "dist", "static-test-chart", "values.yaml"), "utf8");
+      const valuesJSON = yaml.load(valuesYaml) as ValuesJSON;
+      expect(valuesJSON.admission.env).toEqual(expectedAdmissionEnv);
+      expect(valuesJSON.watcher!.env).toEqual(expectedWatcherEnv);
```
Comment (cmwylie19) [inline journey/pepr-build.ts:88]:
```suggestion
      expect(valuesJSON.watcher.env).toEqual(expectedWatcherEnv);
```

File: journey/pepr-build.ts [hunk 2]
```diff
@@ -102,9 +169,9 @@ function parseYAMLToJSON(yamlContent: string): KubernetesObject[] | null {
...
 function sortKubernetesObjects(objects: KubernetesObject[]): KubernetesObject[] {
   return objects.sort((a, b) => {
-      if (a?.kind !== b?.kind) {
-          return (a?.kind ?? '').localeCompare(b?.kind ?? '');
-      }
-      return ((a && a.metadata && (a.metadata as V1ObjectMeta)?.name) ?? '').localeCompare((b && b.metadata && (b.metadata as V1ObjectMeta)?.name) ?? '');
+    if (a?.kind !== b?.kind) {
+      return (a?.kind ?? '').localeCompare(b?.kind ?? '');
+    }
+    return ((a && a.metadata && (a.metadata as V1ObjectMeta)?.name) ?? '').localeCompare((b && b.metadata && (b.metadata as V1ObjectMeta)?.name) ?? '');
```
Comment (cmwylie19) [inline journey/pepr-build.ts:175]: unintentional format, should we start formatting our tests?

File: src/lib/helpers.test.ts [hunk 3]
```diff
@@ -1190,3 +1192,132 @@ describe("validateHash", () => {
...
+    const admissionEnv = { FAV_COLOR: "blue" };
+    const result = mergePkgJSONEnv(admissionEnv, pkgJSON);
+    expect(result).toEqual({
+      FAV_COLOR: "red",
+    });
+  });
+
+  test("if there is no package.json env, return other env", () => {
+    const admissionEnv = { FAV_COLOR: "blue" };
+    const result = mergePkgJSONEnv(admissionEnv);
```
Comment (btlghrants) [inline src/lib/helpers.test.ts]:
```suggestion
    const result = mergePkgJSONEnv(admissionEnv, undefined);
```

**Review by btlghrants — APPROVED** [review]
As talked through, LGTM.

**Pull request merged**